src/main.py

```diff
@@ -1,6 +1,597 @@
+        "cursor": "None",
+    }
+
+    response_json = get_repository_page(logger, ql, variables)
+
+    response_repositories = filter_response(logger, response_json)
+
+    repositories.extend(response_repositories)
+
+    while response_json["data"]["organization"]["repositories"]["pageInfo"]["hasNextPage"]:
+        cursor = response_json["data"]["organization"]["repositories"]["pageInfo"]["endCursor"]
+
+        variables["cursor"] = cursor
+
+        logger.log_info(f"Getting page {number_of_pages + 1} with cursor {cursor}.")
+
+        response_json = get_repository_page(logger, ql, variables)
+
+        response_repositories = filter_response(logger, response_json)
+
+        repositories.extend(response_repositories)
+
+        number_of_pages += 1
+
+    return repositories, number_of_pages
+
+
+def load_archive_rules(archive_rules: dict) -> Tuple[int, int, str, str, int]:
+    """Loads the archive rules from the configuration file.
+
+    Args:
+        archive_rules (dict): The archive rules from the configuration file.
+
+    Returns:
+        Tuple[int, int, str, str, int]: The archive threshold, the notification period, the notification issue tag, the exemption filename, and the maximum number of notifications.
+    """
+    archive_threshold = int(get_dict_value(archive_rules, "archive_threshold"))
+    notification_period = int(get_dict_value(archive_rules, "notification_period"))
+    notification_issue_tag = str(get_dict_value(archive_rules, "notification_issue_tag"))
+    exemption_filename = str(get_dict_value(archive_rules, "exemption_filename"))
+    maximum_notifications = int(get_dict_value(archive_rules, "maximum_notifications"))
+
+    return archive_threshold, notification_period, notification_issue_tag, exemption_filename, maximum_notifications
+
+
+def handle_response(logger: wrapped_logging, response: Any, message: str) -> bool:
+    """Checks if the given response is a valid response object. If not, it logs an error message.
+
+    Args:
+        logger (wrapped_logging): The logger object.
+        response (Any): The response object to check.
+        message (str): The error message to log if the response is not valid.
+
+    Returns:
+        bool: True if the response is valid, False otherwise.
+    """
+    if type(response) is not Response:
+        logger.log_error(
+            message,
+        )
+        return False
+
+    return True
+
+
+def process_repositories(  # noqa: C901
+    interfaces: list[Any],
+    org: str,
+    repositories: list[dict],
+    archive_criteria: list[str],
+    notification_content: list[str],
+) -> Tuple[int, int]:
+
+    issues_created = 0
+    repositories_archived = 0
+
+    logger, rest = interfaces
+    archive_threshold, notification_period, notification_issue_tag, maximum_notifications = archive_criteria
+    notification_issue_title, notification_issue_body = notification_content
+
+    notice_issued = False
+
+    for repository in repositories:
```

Comment:
Just be wary of overly relying on continue, I think it is just about ok here because the logging does a good job of catching where the loops are skipped and guide the reader but it can become difficult to maintain and error prone if the function grows.